### ISSUE TITLE
lambda invoke failed, shell script not found

### ISSUE THREAD

**Issue by jpabbuehl** [localstack/localstack#3518]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description

invoke lambda function (python) using local code as described in the [readme.md](https://github.com/localstack/localstack#using-local-code-with-lambda)

## Expected behavior

should get python response with print statement

## Actual behavior

lambda invoke fails to start
![image](https://user-images.githubusercontent.com/2730222/105891411-e9038f80-6010-11eb-96fc-77636391fad3.png)

But the file exists
![image](https://user-images.githubusercontent.com/2730222/105891592-17816a80-6011-11eb-9920-9ad3b6a081b3.png)

# Steps to reproduce

docker-compose up -d
aws --endpoint-url http://localhost:4566 lambda invoke --function-name test response.json

System is MacOS Catalina, version 10.15.7
docker version 20.10.2
docker-compose version 1.27.4

## Command used to start LocalStack

docker-compose up -d

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

docker-compose.yml

```yml
version: '3.9'
services:
  localstack:
    image: localstack/localstack:0.12.5
    container_name: localstack
    ports:
      - 8055:8080
      - 4566:4566
    environment:
      - SERVICES=s3,lambda
      - DOCKER_HOST=unix:///var/run/docker.sock
      - DEFAULT_REGION=eu-central-1
      - DEBUG=1
      - DATA_DIR=/tmp/localstack/data
      - HOST_TMP_FOLDER=${PWD}/localstack_tmp
      - LAMBDA_REMOTE_DOCKER=false
      - LAMBDA_EXECUTOR=docker
      - LAMBDA_LOCAL=${PWD}/lambda
      - LOCALSTACK_API_KEY=1234
      - MAIN_CONTAINER_NAME=localstack
    volumes:
      - localstack_init:/docker-entrypoint-initaws.d
      - /var/run/docker.sock:/var/run/docker.sock
    networks:
      - custom_network
    healthcheck:
      test:
        - CMD
        - bash
        - -c
        - awslocal s3api wait bucket-exists --bucket test-bucket

networks:
  custom_network:
    driver: bridge

volumes:
  localstack_init:
    driver: local
    driver_opts:
      type: none
      device: ${PWD}/localstack_init
      o: bind
```

lambda/app.py
```
import sys

def handler(event, context):
    return 'Hello from AWS Lambda using Python' + sys.version + '!'
```

localstack_init/lambda.sh
```
#!/bin/bash
set -x
awslocal lambda create-function \
    --function-name test \
    --region eu-central-1 \
    --code S3Bucket="__local__",S3Key="$LAMBDA_LOCAL" \
    --handler app.handler \
    --runtime python3.8 \
    --role whatever
set +x
```

localstack_init/s3.sh
```
#!/bin/bash
set -x
awslocal s3 mb s3://test-bucket
set +x
```

**Comment by mgagliardo** [comment]
Hello @jpabbuehl, I have just tested this and it is working as expected on my machine, my setup.

```
Linux Debian Buster
docker-compose version 1.28.0, build d02a7b1a
Docker version 20.10.2, build 2291f61
Localstack fresh latest image (just pulled)
```

As you can see:
```
$ aws --endpoint-url=http://localhost:4566 --profile=test --region=eu-central-1 lambda invoke --function-name test response.json
$LATEST		200

$ cat response.json 
"Hello from AWS Lambda using Python3.8.6 (default, Nov  4 2020, 20:07:30) \n[GCC 7.3.1 20180712 (Red Hat 7.3.1-9)]!"
```

I *think* you are not passing the parameters correctly, as you can see I am additionally passing region and profile. 
In any case, I will try to test this today on my mac, but in the meantime can you give it a try with this additional params?


Some additional info attached:

```
.
├── docker-compose.yml
├── lambda
│   └── app.py
├── localstack_init
│   ├── lambda.sh
│   └── s3.sh
├── localstack_tmp
└── test.txt
```

`test.txt`
```
"Hello from AWS Lambda using Python3.8.6 (default, Nov  4 2020, 20:07:30) \n[GCC 7.3.1 20180712 (Red Hat 7.3.1-9)]!"
```

Same `docker-compose.yml` and files you have used as you can see:

```
$ awslocal lambda list-functions
{
    "Functions": [
        {
            "FunctionName": "test",
            "FunctionArn": "arn:aws:lambda:eu-central-1:000000000000:function:test",
            "Runtime": "python3.8",
            "Role": "whatever",
            "Handler": "app.handler",
            "Description": "",
            "Timeout": 3,
            "LastModified": "2021-01-26T21:10:43.423+0000",
            "Version": "$LATEST",
            "VpcConfig": {},
            "TracingConfig": {
                "Mode": "PassThrough"
            },
            "RevisionId": "99059953-cd19-47ea-98dd-7cb93f559fe2",
            "State": "Active",
            "LastUpdateStatus": "Successful",
            "PackageType": "Zip"
        }
    ]
}
```
```
$ awslocal s3api list-buckets
{
    "Buckets": [
        {
            "Name": "test-bucket",
            "CreationDate": "2021-01-26T21:10:45.000Z"
        }
    ],
    "Owner": {
        "DisplayName": "webfile",
        "ID": "bcaf1ffd86f41161ca5fb16fd081034f"
    }
}
```

I am using `lambda invoke` to call the function:
```
$ awslocal lambda invoke --function-name test test.txt
{
    "StatusCode": 200,
    "LogResult": "",
    "ExecutedVersion": "$LATEST"
}
```

Lambda logs:
```
localstack    | > START RequestId: 012fb533-911a-1fef-9edd-20cbd695caaa Version: $LATEST
localstack    | > END RequestId: 012fb533-911a-1fef-9edd-20cbd695caaa
localstack    | > REPORT RequestId: 012fb533-911a-1fef-9edd-20cbd695caaa	Init Duration: 363.55 ms	Duration: 3.61 ms	Billed Duration: 100 ms	Memory Size: 1536 MB	Max Memory Used: 25 MB	
localstack    | 2021-01-26T21:22:34:INFO:localstack.services.edge: Unable to find forwarding rule for host "localhost:4566", path "GET /", target header "", auth header "", data "b''"
localstack    | 2021-01-26T21:22:57:WARNING:localstack.services.awslambda.lambda_executors: Empty event body specified for invocation of Lambda "arn:aws:lambda:eu-central-1:000000000000:function:test"
localstack    | 2021-01-26T21:22:57:INFO:localstack.services.awslambda.lambda_executors: Running lambda cmd: docker run -i  -v "/tmp/issue3518/lambda":/var/task -e DOCKER_LAMBDA_USE_STDIN="$DOCKER_LAMBDA_USE_STDIN" -e LOCALSTACK_HOSTNAME="$LOCALSTACK_HOSTNAME" -e EDGE_PORT="$EDGE_PORT" -e _HANDLER="$_HANDLER" -e AWS_LAMBDA_FUNCTION_TIMEOUT="$AWS_LAMBDA_FUNCTION_TIMEOUT" -e AWS_LAMBDA_FUNCTION_NAME="$AWS_LAMBDA_FUNCTION_NAME" -e AWS_LAMBDA_FUNCTION_VERSION="$AWS_LAMBDA_FUNCTION_VERSION" -e AWS_LAMBDA_FUNCTION_INVOKED_ARN="$AWS_LAMBDA_FUNCTION_INVOKED_ARN" -e AWS_LAMBDA_COGNITO_IDENTITY="$AWS_LAMBDA_COGNITO_IDENTITY"   --rm "lambci/lambda:python3.8" "app.handler"
localstack    | 2021-01-26T21:22:58:DEBUG:localstack.services.awslambda.lambda_executors: Lambda arn:aws:lambda:eu-central-1:000000000000:function:test result / log output:
localstack    | "Hello from AWS Lambda using Python3.8.6 (default, Nov  4 2020, 20:07:30) \n[GCC 7.3.1 20180712 (Red Hat 7.3.1-9)]!"
localstack    | > START RequestId: 7772f1bf-6df5-1d21-ff54-2f680fdd4250 Version: $LATEST
localstack    | > END RequestId: 7772f1bf-6df5-1d21-ff54-2f680fdd4250
localstack    | > REPORT RequestId: 7772f1bf-6df5-1d21-ff54-2f680fdd4250	Init Duration: 237.68 ms	Duration: 3.78 ms	Billed Duration: 100 ms	Memory Size: 1536 MB	Max Memory Used: 25 MB	
```

**Comment by jpabbuehl** [comment]
thanks for your reply. I helped

After running docker image prune, pulling latest image manually (instead of the image with the most recent tag) and change docker-compose with localstack/localstack:latest, the issue is now different !

When running:
```
aws --endpoint-url=http://localhost:4566 --profile=test --region=eu-central-1 lambda invoke --function-name test response.json
```

I am getting the following
![image](https://user-images.githubusercontent.com/2730222/105951946-20098d80-6071-11eb-8a94-800b8996d36c.png)

the module app is not found. I understood from the docs that you could you lambda code without putting them in a zip.

**Comment by whummer** [comment]
Thanks for reporting @jpabbuehl . When using the `LAMBDA_REMOTE_DOCKER=false`, can you please ensure that the `HOST_TMP_FOLDER` variable is set properly, and that a volume mount is added to your docker-compose section to enable mounting of the local folder into the container:
```
volumes:
   ...
   - ${PWD}/localstack_tmp:/tmp/localstack
```

See also this section in the README: https://github.com/localstack/localstack#using-local-code-with-lambda Hope that helps - please keep us posted. Thanks

**Comment by jpabbuehl** [comment]
Thanks @whummer and @mgagliardo 

Adding ${PWD}/localstack_tmp:/tmp/localstack  in docker-compose.yml didn't solve the issue :(

I have tested the same code on another linux machine and I get the same issue, module app not found

```
docker version: 20.10.2
docker-compose version: 1.27.4
OS: Pop!_OS 20.04
localstack image id: e59dec65ed80 (tag latest)
```

**Comment by mgagliardo** [comment]
Hello @jpabbuehl, so it seems there has been a bug introduced in the `latest` image, please go back to my comment and just change the `latest` for release `0.12.5`  and test it out. I have just tried on a Linux and MacOS (Big Sur) and it is working as expected.

I will take care of debugging this then, please confirm if it worked for you. Cheers.

**Comment by jpabbuehl** [comment]
@mgagliardo  I confirm using 0.12.5 and adding    - ${PWD}/localstack_tmp:/tmp/localstack from @whummer  solves the issue on both linux and macOS.

Thanks for your work on localstack and your help.

**Comment by mgagliardo** [comment]
Thanks for the prompt response @jpabbuehl, we are working on a fix to get this solved on the latest release ASAP. Closing this one. Cheers!

**Comment by whummer** [comment]
Thanks for the update @jpabbuehl , this should be fixed in 7a90cf4 . Can you please pull the `latest` Docker image and give it another try? Please keep us posted how it goes, thanks!

**Comment by jpabbuehl** [comment]
hi @whummer. I confirm the issue is resolved when pulling latest image (sha 29449f457169)
